go-controller/pkg/util/net.go

```diff
@@ -389,3 +390,32 @@ func RouteEqual(l, r *netlink.Route) bool {
 		l.Congctl == r.Congctl &&
 		l.FastOpenNoCookie == r.FastOpenNoCookie
 }
+
+// SubnetBroadcastIP returns the IP network's broadcast IP.
+func SubnetBroadcastIP(ipnet net.IPNet) net.IP {
+	// For IPv4.
+	if ipnet.IP.To4() != nil {
+		// ip address in uint32
+		ipBits := binary.BigEndian.Uint32(ipnet.IP.To4())
+		// mask will give us all fixed bits of the subnet
+		maskBits := binary.BigEndian.Uint32(ipnet.Mask)
+		// inverted mask will give us all moving bits of the subnet
+		invertedMaskBits := maskBits ^ 0xffffffff // xor the mask
+		// network ip
+		networkIPBits := ipBits & maskBits
+		// broadcastIP = networkIP added to the inverted mask
+		broadcastIPBits := networkIPBits | invertedMaskBits
+		broadcastIP := make(net.IP, 4)
+		binary.BigEndian.PutUint32(broadcastIP, broadcastIPBits)
+		return broadcastIP
+	}
+	// For IPv6. This conversion is actually easier, it follows the same principle as above.
+	byteIP := []byte(ipnet.IP)                // []byte representation of IP
+	byteMask := []byte(ipnet.Mask)            // []byte representation of mask
+	byteTargetIP := make([]byte, len(byteIP)) // []byte holding target IP
+	for k := range byteIP {
+		invertedMask := byteMask[k] ^ 0xff // inverted mask byte
+		byteTargetIP[k] = byteIP[k]&byteMask[k] | invertedMask
+	}
+	return net.IP(byteTargetIP)
```

Comment:
nit: can we move this into an helper and invoke it from here ?

Reply:
I don't think this helper would be useful elsewhere, I would prefer to keep it as is.

Reply:
even if the helper is used in one place, a properly named helper helps alleviate the burden on a casual reader. Too many details now. 

But OK, your call.